### DISCUSSION TITLE
Update Apps from new template versions

### DISCUSSION THREAD

**Discussion by peibolsang** [vercel/community#555] [Idea]
Hi team,

I am wondering if the following functionality makes sense at Vercel:

- Let's say I have created an application from a template, using a concrete template version (snapshot).
- I made my customizations to the application, adding my own stuff and even removing some boilerplate content I don't need. 
- After weeks of development, Vercel updates the master template adding new cool features and deprecating others.

Does it make sense that Vercel gives me some sort of functionality to update my app using the new/latest version of the app **auto-magically**? I am unsure and I can think of a few challenges (ie I may have turned the code upside down) ... but I am evaluating options that help me update template-based Web apps so they are always on the latest and greatest versions of the providers (eg Vercel).

Do you think this is even possible?

Thank you!

**Comment by swarnava** [comment]
Hi there,

With new template changes, it won't happen automatically in your codebase since Vercel actually fork the template repo and create a separate repo on your account. You need to manually update it when there is a new release. Thanks!

**Comment by peibolsang** [comment]
Thanks for the answer @swarnava. Could you elaborate more on the solution? What do you mean that Vercel forks the template repo and creates a separate repo on my account?

Thank you!

**Comment by swarnava** [comment]
I am assuming you are referring to something like this: https://github.com/vercel/vercel/tree/main/examples/hugo. When you deploy it in Vercel, you will create an own forked version: 

![image](https://user-images.githubusercontent.com/1162991/162960281-bb433222-6648-47e7-86d4-82305ac6c7bc.png)

Once you do that, it's up to you to maintain the forked version since it's entirely a separate source code and won't remain align with the source template.

**Reply by peibolsang** [reply]
Perfect answer! This is exactly what I wanted to know.

Thank you
